rflx/model.py

```diff
@@ -49,13 +52,29 @@ def __repr__(self) -> str:
 
 
 class Type(Base):
-    def __init__(self, identifier: StrID) -> None:
+    def __init__(
+        self, identifier: StrID, location: Location = None, error: RecordFluxError = None
+    ) -> None:
         identifier = ID(identifier)
-
-        if len(identifier.parts) != 2:
-            raise ModelError(f'unexpected format of type name "{identifier}"')
+        self.error = error or RecordFluxError()
+
+        if len(identifier.parts) != 2 and not str(identifier.name).startswith("__REFINEMENT__"):
```

Comment:
I changed this in 4b2ae3a1bcd96c2422f0a578ea2b5e2d77860f6d. Here is the explanation form the commit message:

> When using an invalid package name for a refinement (say "A.B"), then
    the expected error is supposed to be something like "A.B is not a valid
    package name". For refinements, however, the super class is initialized
    with

```python
package * "__REFINEMENT__"
f"{flat_name(sdu.full_name)}__{flat_name(pdu.full_name)}__{field.name}__"
```
    
>    which produces some convoluted error message mentioning that long
    internal string. Instead of filtering such situations in the super-class,
    we now reinitialized the error buffer after initializing the super-class
    of Refinement, effectively ignoring errors in the ID() constructor. The
    well-formedness of the package identifier is checked separately.